runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -365,21 +352,15 @@ private boolean validateQueryParams(
     {
         Matcher matcher = requestType.queryMatcher.reset(path);
         boolean valid = true;
-        boolean found = matcher.find();
-        while (found)
+        while (valid && matcher.find())
         {
             String name = matcher.group(1);
             Validator validator = requestType.queryParams.get(name);
             if (validator != null)
             {
                 String8FW value = new String8FW(matcher.group(2));
-                if (!validator.read(value.value(), value.offset(), value.length()))
-                {
-                    valid = false;
-                    break;
-                }
+                valid = validator.read(value.value(), value.offset(), value.length());
```

Comment:
This should be `valid &= ...` to better convey the intent.

Reply:
OK